nextflow/EVAImport/main.nf

```diff
@@ -0,0 +1,301 @@
+}
+
+if( (!params.old_host || !params.old_port || !params.old_dbname) && params.species == "rattus_norvegicus") {
+  exit 1, "ERROR: please provide a host (--old_host), port (--old_port) and db name (--old_dbname) for a previous rat database"
+}
+
+if(!params.output_file) {
+  exit 1, "ERROR: please provide an output file to the EVA import script"
+}
+
+// Build command to run EVA import
+command_to_run = " -i ${params.input_file} --source ${params.source} --source_description '${params.description}' --version ${params.version} --registry ${params.registry} --species ${params.species} --skip_tables '${params.skip_tables}'"
+
+log.info """
+  Importing EVA data with the following parameters: \
+  ${command_to_run}
+"""
+
+process run_eva {
+  cpus "${params.fork}"
+
+  input:
+  val wait
+  path eva_script
+  val options
+  val merge_all_types
+  val fork
+  val sort_vf
+  val chr_synonyms
+  val remove_prefix
+  val output_file
+  
+  output: val 'ok'
+  
+  script:
+  def sort_vf_table     = sort_vf ? " --sort_vf" : ""
+  def merge_all         = merge_all_types ? " --merge_all_types" : ""
+  def use_fork          = fork ? "--fork ${fork}" : ""
+  def chr_synonyms_file = chr_synonyms ? " --chr_synonyms ${chr_synonyms}" : ""
+  def rm_prefix         = remove_prefix ? " --remove_prefix ${remove_prefix}" : ""
+  
+  """
+  perl ${eva_script} ${options} $sort_vf_table $merge_all $use_fork $chr_synonyms_file $rm_prefix --output_file ${output_file}
+  """
+}
+
+process run_variant_synonyms {
+  input:
+  val wait
+  path var_syn_script
+  val source_name
+  val species
+  val input_file
+  val registry
+  val host
+  val port
+  val dbname
+  
+  output: val 'ok'
+  
+  script:
+  
+  if(species == "sus_scrofa")
+      """
+        perl ${var_syn_script} --source_name ${source_name} --species ${species} --data_file ${input_file} --registry ${registry}
+        perl ${var_syn_script} --source_name "pig_chip" --species ${species} --registry ${registry}
+      """
+  
+  else if(species == "rattus_norvegicus")
+      """
+        perl ${var_syn_script} --source_name ${source_name} --species ${species} --data_file ${input_file} --registry ${registry}
+        perl ${var_syn_script} --source_name "rat" --species ${species} --registry ${registry} --host ${host} --port ${port} --user 'ensro' --db_name $dbname
+      """
+  else 
+      """
+        perl ${var_syn_script} --source_name ${source_name} --species ${species} --data_file ${input_file} --registry ${registry}
+      """
+}
+
+process run_variation_set {
+  input:
+  val wait
+  val var_set_script
+  val files_path
+  val filenames
+  val set_names
+  val species
+  val registry
+
+  output: val 'ok'
+
+  exec:
+
+  // variable my_set has to be before any if statement
+  // related to this issue: https://github.com/nextflow-io/nextflow/issues/804
+  def my_set = set_names.get(species)
+  for (String name : my_set) {
```

Comment:
Does it make sense to load the sets in parallel from db load perspective? Can we use [Channels](https://www.nextflow.io/docs/latest/channel.html#fromlist) to do the same?

Reply:
The variant sets only start when variation synonyms is finished.